### PR TITLE
Add a Shellckecker to Travis Jobs

### PR DESCRIPTION
**Pull request by Tatarinho**
There isn't currently any Shellcheck in the Travis Jobs for the
Fujitsu/monasca-docker. The purpose of this commit is to add proper
stages to .travis configuration file and fix all shell warnings/errors
in the code.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,13 +23,15 @@ before_install:
   - docker --version
   - docker-compose --version
   - ifconfig
-  - eth0_ipaddr=$(ifconfig eth0 | awk '{ print $2}' | grep -E -o "([0-9]{1,3}[\.]){3}[0-9]{1,3}")
+  - eth0_ipaddr=$(ifconfig docker | awk '{ print $2}' | grep -E -o "([0-9]{1,3}[\.]){3}[0-9]{1,3}")
```
Comment (Tatarinho) [inline .travis.yml]: Not sure about it. There is no eth0, but I wasn't sure if it's ok.

File: monasca-python/build.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ else
               if echo "$extra" | grep -q "=="
               then
                 pip install --no-cache-dir "$extra"
-                dep=$(echo "$extra" | cut -d'=' -f1)
+                # dep=$(echo "$extra" | cut -d'=' -f1)
```
Comment (Tatarinho) [inline monasca-python/build.sh]: dep is not used (?)

File: monasca-python/build.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@ set -x
 
 if [ -e /apk.sh ]; then
     echo "Overriding default apk dependencies install function"
-    source /apk.sh
```
Comment (matrixik) [inline monasca-python/build.sh]: Are you sure it's working fine without `source`?

Maybe try using `shellcheck -x` or `# shellcheck source=somefile`
https://github.com/koalaman/shellcheck/wiki/SC1091

Same comment to all places you removed `source`.

File: monasca-python/build.sh [hunk 4]
```diff
@@ -14,13 +14,13 @@ fi
 
 if [ -e /install.sh ]; then
     echo "Overriding default install function"
-    source /install.sh
+    /install.sh
 else
     echo "Using default install function"
     install() {
-        local constraints=${1}
```
Comment (matrixik) [inline monasca-python/build.sh]: Why are you exporting this variables?

File: monasca-python/build.sh (hunk 2 above)
Comment (matrixik) [reply monasca-python/build.sh]: Possible, then I think it can be removed.

File: monasca-python/build.sh [hunk 5]
```diff
@@ -52,12 +52,13 @@ fi
...
 if [ -e /clone.sh ]; then
     echo "Overriding default clone function"
-    source /clone.sh
+    # shellcheck disable=SC1091
+    /clone.sh
 else
     echo "Using default clone function"
     clone() {
-        local repo=$1
-        local branch=$2
```
Comment (matrixik) [inline monasca-python/build.sh]: Why exporting?

File: monasca-python/build.sh [hunk 6]
```diff
@@ -125,4 +126,4 @@ echo "Installing ${REPO}" && install "${CONSTRAINTS}" "${EXTRAS}" "${EXTRA_DEPS}
 cd - || exit
 rm -rf /install.sh /clone.sh /install_apk_deps.sh /app /root/.cache/pip
 
-apk del build-dep
```
Comment (matrixik) [inline monasca-python/build.sh]: Why empty line removed from end of the file?

File: .travis.yml (hunk 1 above)
Comment (matrixik) [reply .travis.yml]: So this variable name should be changed.

And did you check how it's working now? I'm talking about next line `echo "Replacing ip addresses in .env file with dynamic travis eth0 ip:" $eth0_ipaddr`

**Pull request closed without merging**